tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/BigIntElement.java

```diff
@@ -0,0 +1,36 @@
+package dk.alexandra.fresco.tools.ot.base;
+
+import java.math.BigInteger;
+
+public class BigIntElement implements InterfaceNaorPinkasElement {
+
+  BigInteger element;
+  BigInteger dhModulus;
+
+  public BigIntElement(BigInteger element, BigInteger dhModulus) {
+    this.element = element;
+    this.dhModulus = dhModulus;
+  }
+
+  @Override
+  public byte[] toByteArray() {
+    return this.element.toByteArray();
+  }
+
+  @Override
+  public InterfaceNaorPinkasElement groupOp(InterfaceNaorPinkasElement other) {

```

Comment:
I think this could be done more elegantly by adding a generic parameter to InterfaceNaorPinkasElement to avoid type casting

Reply:
Now uses templates, no cast needed anymore